R/lmer_multimember.R

```diff
@@ -380,113 +370,97 @@ lme4::fixef
 #' @description The \code{glmerModMultiMember} class extends \code{glmerMod}
-#' (which extends \code{merMod}) from the \pkg{lme4}-package.
+#' from the \pkg{lme4}-package and \code{merModMultiMember}.
 #' @seealso \code{\link[lme4]{glmer}} and \code{\link[lme4]{merMod}}
 #' @export
 #' @importClassesFrom lme4 glmerMod
 #' @return An object of class \code{glmerModMultiMember} similar to
-#' \code{glmerMod} objects (see \code{\link[lme4]{merMod}}) with
-#' extra information about the multimembership random effects.
+#' \code{merModMultiMember} objects but inheriting from \code{glmerMod}
 glmerModMultiMember <-
   setClass("glmerModMultiMember",
-    contains = c("glmerMod"), slots = c(memberships = "list")
+           contains = c("glmerMod", "merModMultiMember")
   )
 
 
-#' @title summary method for multimembership lmer model objects
-#' @param object lmerModMultiMember model object
-#' @param ... additional arguments to be passed on to summary.lmerMod
-#' @return summary of lmerModMultiMember object
+#' @title Summary method for multimembership model objects
+#' @param object merModMultiMember model object
+#' @param ... additional arguments to be passed on to summary.merMod
+#' @return summary of merModMultiMember object
 #' @export
-summary.lmerModMultiMember <- function(object, ...) {
-  if (!inherits(object, "lmerModMultiMember") && !inherits(object, "lmerMod")) {
+summary.merModMultiMember <- function(object, ...) {
+  if (!inherits(object, "merModMultiMember") && !inherits(object, "merMod")) {
     stop(
       "Cannot compute summary for objects of class: ",
       paste(class(object), collapse = ", ")
     )
   }
 
-  memberships <- object@memberships
-  summ <- summary(as(object, "lmerMod"), ...)
-  summ$objClass <- class(object) # Used by lme4:::print.summary.lmerMod
-  summ$methTitle <- paste0(
-    summ$methTitle,
-    " with multiple membership random effects"
-  )
-  summ$memberships <- memberships
-  class(summ) <- c("summary.lmerModMultiMember", class(summ))
-
-  return(summ)
-}
-
-
-#' @title summary method for multimembership glmer model objects
-#' @param object glmerModMultiMember model object
-#' @param ... additional arguments to be passed on to summary.glmerMod
-#' @return summary of glmerModMultiMember object
-#' @export
-summary.glmerModMultiMember <- function(object, ...) {
-  if (!inherits(object, "glmerModMultiMember") &&
-    !inherits(object, "glmerMod")) {
-    stop(
-      "Cannot compute summary for objects of class: ",
-      paste(class(object), collapse = ", ")
-    )
+  # check whether model was lmer or glmer and call summary
+  # if neither lmer or glmer throw error
+  if (inherits(object, "lmerMod")) {
+    summ <- summary(as(object, "lmerMod"), ...)
+  } else {
+    if (inherits(object, "glmerMod")) {
+      summ <- summary(as(object, "glmerMod"), ...)
+    } else {
+      stop("Object class must inherit from either lmerMod or glmerMod")
+    }
```

Comment:
This if/else tree is a little ugly, but I think the alternative would be to have separate `summary` functions for `lmerModMultiMember` and `glmerModMultiMember` which is a little redundant.

Reply:
There are other ways to achieve this, but it's not worth either of our time right now to rewrite something that works.